vars/provision.md

```diff
@@ -0,0 +1,25 @@
+# provision
+
+In order to use it with the default conventions simply add the following to
+your Jenkinsfile.
+For the details of what happens please review the
+[provision](provision.groovy) script.
+
+## Prerequisite
+For dynamic resources you will need a `provision.azure.yaml` file in your repo with the below format.
```

Comment:
I would break this down a bit so it's clearer what the file needs to look like.

State that the repo needs a `provision.azure.yaml` file in the root directory and just show the `resources` content.

Then have a separate section describing the extra content needed for queues.  Add an example there that shows just a single queue and make it clear that they should define their own queue name.  You can use say `claim` as an example if you think it makes it easier to understand.